### ISSUE TITLE
EntityStorageException: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'bd.field_data_field_image' doesn't exist in NodeStorageController->save() (line 304 of C:\xampp\htdocs\BD\core\modules\node\node.entity.inc)

### ISSUE THREAD

**Issue by ColumbiaBlooms** [backdrop/backdrop-issues#2621]
Did some web searches for the EntityStorageException message I get when trying the procedure at the bottom of https://backdropcms.org/user-guide/deep-dive-advanced-layout-options  (adding an image field block). All I found were old (a year or more ago) reports for this problem. Nothing seemed to fit my situation. Any ideas as to how to fix this? Maybe I misunderstood the instructions. Since I already have a Post content type I created a new node of this type. I then clicked the choose file button (under the body section) and uploaded an image. I then clicked Save and Boom! I get the error message "EntityStorageException: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'bd.field_data_field_image' doesn't exist in NodeStorageController->save() (line 304 of C:\xampp\htdocs\BD\core\modules\node\node.entity.inc)."

**Comment by klonos** [comment]
Lets note that this happens in a site upgraded from D7.

It seems like this is error is saying that it cannot find a table in the database that it expects to be there. Will have to try to reproduce this by doing the following:

1. Install a fresh D7 site.
2. Upgrade to Backdrop.
3. Try to create a post node with an image file.
4. See if that causes the error.

1. Install a fresh D7 site.
2. Create an article with an image file. (**additional step**)
3. Upgrade to Backdrop.
4. Try to create a post node with an image file.
5. See if that causes the error.

**Comment by klonos** [comment]
From my googling around, it seems that other people having similar problems have also been migrating from D5/D6 to D7 and some tables have not been created/updated properly due to some errors along the process.

**Comment by Graham-72** [comment]
I experienced something like this and it proved to be caused by PHP not having a large enough memory allocation on my server. I increased the PHP settings from 128M to 192M and no more problems - but this may have nothing to do with your problem.

**Comment by ColumbiaBlooms** [comment]
@Graham-72 My `memory_limit=512M` in C:\xampp\php\php.ini. Is that the setting you are referring to?

@klonos Do you want me to go through the process of reproducing this problem? It's kinda unclear from your post ...

**Comment by Graham-72** [comment]
@ColumbiaBlooms Another way to see your current PHP memory limit is to look at th Backdrop Status Report page for the line 'PHP Memory Limit'. If this is 512M then this is not the same problem as mine.

**Comment by ColumbiaBlooms** [comment]
@Graham-72 Backdrop Status report confirms 512M.

**Comment by klonos** [comment]
@ColumbiaBlooms I was just taking notes of what needs to be tested in order for anybody to reproduce (or not) this issue. All was general and not intended as instructions for you to do something. One of us will have to test migration from D7 and verify whether this is an actual issue in that case. Perhaps all we need to do is just create that table during upgrade.

**Comment by laryn** [comment]
I think I just ran into this, or very similar. 

I believe it was due to me trying to delete the (unused) Post content type and the respective image field. I've worked this out over a few days and may not be remembering all the details accurately, but I thought I had deleted it completely. Today the Post type was back (but without a `field_image`). I added `field_image` again via the UI and now these errors are gone. Are the `post` content type and `field_image` hard coded somewhere?

**Comment by laryn** [comment]
No, I just deleted the content type on a sandbox site and ran cron without getting a flood of these types of errors in the log. Hmmm..

**Comment by laryn** [comment]
I deleted the Post type again on the upgraded/dev site and then got an error about `field_tags`. I deleted a few related `field_tags` config files (I know...) and reran cron and don't see the errors anymore. 

Notably, this site has come through a number of Drupal upgrade processes over the years (maybe even started in Drupal 5) and some of the field names are definitely non-standard. e.g. most content types don't have `field_tags` but `field_taxonomy_vocabulary_x`. Not sure when those non-standards crept in or if that plays into it...
